pkg/controllers/provisioning/provisioner.go

```diff
@@ -372,14 +373,185 @@ func (p *Provisioner) getDaemonSetPods(ctx context.Context) ([]*v1.Pod, error) {
 	if err := p.kubeClient.List(ctx, daemonSetList); err != nil {
 		return nil, fmt.Errorf("listing daemonsets, %w", err)
 	}
+
+	limitRangeList := &v1.LimitRangeList{}
+	if err := p.kubeClient.List(ctx, limitRangeList); err != nil {
+		return nil, fmt.Errorf("listing daemonsets, %w", err)
+	}
+
+	limitRageMap := map[string][]v1.LimitRange{}
+	for _, lr := range limitRangeList.Items {
+		limitRageMap[lr.Namespace] = append(limitRageMap[lr.Namespace], lr)
+	}
+
 	var ret []*v1.Pod
 	for _, daemonSet := range daemonSetList.Items {
-		pod := &v1.Pod{Spec: daemonSet.Spec.Template.Spec}
-		ret = append(ret, pod)
+		pod := &v1.Pod{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: daemonSet.Namespace},
+			Spec: *daemonSet.Spec.Template.Spec.DeepCopy()}
+		if applyLR(limitRageMap, pod) == nil {
+			ret = append(ret, pod)
+		}
 	}
 	return ret, nil
 }
 
+// Check and apply limitRange configuration on DaemonSet
+func applyLR(limitRanges map[string][]v1.LimitRange, pod *v1.Pod) error {
+	namespaceedLimitRanges, namespaceExists := limitRanges[pod.ObjectMeta.Namespace]
+	if !namespaceExists {
+		return nil
+	}
+
+	// LimitRanges for Containers
+	for _, lr := range namespaceedLimitRanges {
+		for _, limitrangeItem := range lr.Spec.Limits {
+			if limitrangeItem.Type == v1.LimitTypePod {
+				if err := applyLRTypePod(lr.Name, limitrangeItem, pod); err != nil {
+					return err
+				}
+			} else {
+				if err := applyLRTypeContainer(lr.Name, limitrangeItem, pod); err != nil {
+					return err
+				}
+			}
+		}
+	}
+	return nil
+}
+
+// Limit Ranges for Pods
+func applyLRTypePod(name string, LRItem v1.LimitRangeItem, pod *v1.Pod) error {
+	podTotalLimits := resources.LimitsForPods(pod)
+	podTotalRequests := resources.RequestsForPods(pod)
+	delete(podTotalLimits, "pods")
+	delete(podTotalRequests, "pods")
+
+	if !resources.Fits(podTotalLimits, LRItem.Max) && LRItem.Max != nil {
```

Comment:
I can't find the behavior of LimitRange and DaemonSets described in the K8s docs. Can you do some experiments to confirm that:
- If the DS resource request is less than the limit range max, the DS pod won't get created?
- Same test for min & ratio.

Reply:
I don't think unit tests will work.  We need a real kube-controller-manager to create the DS pods.  Can you verify and document this behavior with your cluster?